lib/rules/await-async-query.ts

```diff
@@ -46,10 +48,16 @@ export default createTestingLibraryRule<Options, MessageIds>({
     }
 
     return {
-      'CallExpression Identifier'(node: TSESTree.Identifier) {
```

Comment:
The problem was here. This is getting an `Identifier` n levels within a `CallExpression`, which can lead to an `Identifier` not related at all to the `CallExpression`. Fixed just using `getDeepestIdentifierNode` so we get the Identifier of the `CallExpression` we are interested in.